backend/controllers/appointments/viewAppointments.ts

```diff
@@ -66,13 +67,20 @@ export default ({
 
     const videoLinkAppointments = await bookAVideoLinkApi.getPrisonSchedule(systemContext, agencyId, searchDate)
 
+    const locationMappings = await Promise.all(
+      [...new Set(videoLinkAppointments.map((vlb) => vlb.dpsLocationId))].map((id) =>
+        nomisMapping.getNomisLocationMappingByDpsLocationId(systemContext, id)
+      )
+    )
```

Comment:
Its one call for every VLB appointment in the list? But it does get the unique key. I was thinking could cut a couple of corners and use the `prisonLocDesc` to match against the `locationDescription` - they both hold what the locations API calls the `localName`.

Reply:
Might be an issue if the numbers of VLBs (and later, VLPMs) is very long. Is there a mapping POST endpoint which takes a list of IDs?

Reply:
Yes, that's fine - go for it your way and let's see how it performs. If you look at Sunday 18th Jan in Birmingham - I created two appointments / VLBs that were being mixed up.